drivers/counter/counter_mchp_xec.c

```diff
@@ -90,49 +107,40 @@ static int counter_xec_set_alarm(struct device *dev, u8_t chan_id,
 				 const struct counter_alarm_cfg *alarm_cfg)
 {
 	BTMR_Type *counter = COUNTER_XEC_REG_BASE(dev);
-	const struct counter_xec_config *counter_cfg = COUNTER_XEC_CONFIG(dev);
 	struct counter_xec_data *data = COUNTER_XEC_DATA(dev);
-	u32_t ticks;
 
 	if (chan_id != 0) {
 		LOG_ERR("Invalid channel id %u", chan_id);
 		return -ENOTSUP;
 	}
 
-	if (!(counter->CTRL & MCHP_BTMR_CTRL_ENABLE)) {
-		return -EIO;
+	/* Interrupts are only triggered when the counter reaches 0.
+	 * So only relative alarms are supported.
+	 */
+	if (alarm_cfg->flags & COUNTER_ALARM_CFG_ABSOLUTE) {
+		return -ENOTSUP;
 	}
 
-	if (counter->CTRL & MCHP_BTMR_CTRL_START) {
+	if (data->alarm_cb != NULL) {
 		return -EBUSY;
```

Comment:
Shouldn't this be -EINVAL
* @retval -EINVAL if alarm settings are invalid.
 * @param callback Callback called on alarm (cannot be NULL).

Reply:
Other drivers are returning `-EBUSY` when alarm callback is already set. I am just following them.

Reply:
Got it, I think an addition to API doc indicating -EBUSY in this context will be good